mmv1/third_party/terraform/data_sources/data_source_dns_managed_zone.go

```diff
@@ -1,101 +1,151 @@
 package google
 
 import (
+	"context"
 	"fmt"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"google.golang.org/api/dns/v1"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 )
 
-func dataSourceDnsManagedZone() *schema.Resource {
-	return &schema.Resource{
-		Read: dataSourceDnsManagedZoneRead,
+var _ datasource.DataSource = &GoogleDnsManagedZoneDataSource{}
+
+func NewGoogleDnsManagedZoneDataSource() datasource.DataSource {
+	return &GoogleDnsManagedZoneDataSource{}
+}
+
+// GoogleDnsManagedZoneDataSource defines the data source implementation
+type GoogleDnsManagedZoneDataSource struct {
+	client  *dns.Service
+	project types.String
+}
+
+type GoogleDnsManagedZoneModel struct {
+	Id            types.String `tfsdk:"id"`
+	DnsName       types.String `tfsdk:"dns_name"`
+	Name          types.String `tfsdk:"name"`
+	Description   types.String `tfsdk:"description"`
+	ManagedZoneId types.Int64  `tfsdk:"managed_zone_id"`
+	NameServers   types.List   `tfsdk:"name_servers"`
+	Visibility    types.String `tfsdk:"visibility"`
+	Project       types.String `tfsdk:"project"`
+}
+
+func (d *GoogleDnsManagedZoneDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_managed_zone"
+}
+
+func (d *GoogleDnsManagedZoneDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Provides access to a zone's attributes within Google Cloud DNS",
 
-		Schema: map[string]*schema.Schema{
-			"dns_name": {
-				Type:     schema.TypeString,
+		Attributes: map[string]schema.Attribute{
+			"dns_name": schema.StringAttribute{
 				Computed: true,
 			},
 
-			"name": {
-				Type:     schema.TypeString,
+			"name": schema.StringAttribute{
 				Required: true,
```

Comment:
Do we need `MarkdownDescription` for the attributes?

I guess not, as the docs for this data source are handwritten, but I think it'd be great to include `MarkdownDescription` from the start of using the plugin framework as it makes any automatic docs generation easier to do in future